application/data/standardisers/preset_search.py

```diff
@@ -0,0 +1,266 @@
+    these determine how charts and tables should be displayed. See the categorisation dashboard for examples
+
+    PresetSearch first converts to ethnicity labels from the Race Disparity Audit standard list
+    Then it searches our preset library for possible matches from known categorisations
+    """
+
+    def build_presets_data(self, raw_ethnicities):
+        valid_presets = self.__get_valid_presets(raw_ethnicities)
+
+        preset_data = [preset.get_outputs(raw_ethnicities, self.standardiser) for preset in valid_presets]
+        custom_data = Preset.get_custom_data_outputs(raw_ethnicities)
+
+        all_output_data = preset_data + [custom_data]
+
+        return all_output_data
+
+    def __get_valid_presets(self, raw_ethnicities):
+        return self.preset_collection.get_valid_presets(raw_ethnicities, self.standardiser)
+
+    def __init__(self, standardiser, preset_collection):
+        self.standardiser = standardiser
+        self.preset_collection = preset_collection
+
+
+class Standardiser:
+    def __init__(self, ethnicity_map=None):
+        if ethnicity_map:
+            self.ethnicity_map = ethnicity_map
+        else:
+            self.ethnicity_map = {}
+
+    def __len__(self):
+        return len(self.ethnicity_map)
+
+    def add_conversion(self, raw_ethnicity, standard_ethnicity):
+        lookup_value = self.simplify_key(raw_ethnicity)
+        self.ethnicity_map[lookup_value] = standard_ethnicity
+
+    @classmethod
+    def simplify_key(cls, key_value):
+        return key_value.strip().lower()
+
+    def standardise(self, raw_ethnicity):
+        lookup_value = self.simplify_key(raw_ethnicity)
+        if lookup_value in self.ethnicity_map:
+            return self.ethnicity_map[lookup_value]
+        else:
+            return raw_ethnicity
+
+    def standardise_all(self, raw_ethnicity_list):
+        return [self.standardise(raw_ethnicity) for raw_ethnicity in raw_ethnicity_list]
+
+
+class PresetCollection:
+    def __init__(self):
+        self.presets = []
+
+    def add_preset(self, preset):
+        self.presets.append(preset)
+
+    def get_valid_presets(self, raw_ethnicity_list, preset_standardiser):
+        valid_presets = [
+            preset
+            for preset in self.presets
+            if preset.is_valid_for_raw_ethnicities(raw_ethnicity_list, preset_standardiser)
+        ]
+        valid_presets.sort(key=lambda preset: -preset.get_data_fit_level(raw_ethnicity_list, preset_standardiser))
+        return valid_presets
+
+
+class Preset:
+    def is_valid_for_raw_ethnicities(self, raw_ethnicities, preset_standardiser):
+        standard_ethnicities_in_data = preset_standardiser.standardise_all(raw_ethnicities)
+
+        return self.is_valid_for_standard_ethnicities(standard_ethnicities_in_data)
+
+    def is_valid_for_standard_ethnicities(self, standard_ethnicities):
+        unique_ethnicities = Preset.__remove_duplicates(standard_ethnicities)
```

Comment:
Refactor this to a `__unique_ethnicities()` private instance method rather than a private static class method?